src/main/java/org/rumbledb/runtime/operational/TreatIterator.java

```diff
@@ -32,7 +32,7 @@
 
     private Item nextResult;
     private Item currentResult;
-    private int childIndex;
+    private int resultCount;
```

Comment:
I have renamed this variable to resultCount as I strongly believe that childIndex is a misleading name here. Within the context of iterators child refers to a child iterator. However, this index was merely counting the results that are found so far.

Reply:
great, thanks!